src/Sylius/Bundle/ResourceBundle/Doctrine/ODM/PHPCR/EventListener/DefaultPathListener.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Sylius\Bundle\ResourceBundle\Doctrine\ODM\PHPCR\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormEvent;
+use Doctrine\ODM\PHPCR\DocumentManagerInterface;
+use PHPCR\Util\NodeHelper;
+use Sylius\Component\Resource\Metadata\Registry;
+use Doctrine\ODM\PHPCR\Mapping\ClassMetadata;
+use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+
+/**
+ * Automatically set the parent brefore the creation.
+ *
+ * TODO: Allow `rewrite` (or similar) option to allow the forceful setting
+ *       of the parent even if the parent is already set (and is different from
+ *       the candidate parent).
+ * TODO: Only applies when ClassMetadata::GENERATOR_TYPE_PARENT strategy is used.
+ *
+ * @author Daniel Leech <daniel@dantleech.com>
+ */
+class DefaultPathListener
+{
+    /**
+     * @var Registry
+     */
+    private $registry;
+
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $documentManager;
+
+    /**
+     * @param Registry $registry
+     * @param DocumentManagerInterface $documentManager
+     */
+    public function __construct(
+        Registry $registry,
+        DocumentManagerInterface $documentManager
+    )
+    {
+        $this->registry = $registry;
+        $this->documentManager = $documentManager;
+    }
+
+
+    public function onPreCreate(ResourceControllerEvent $event)
+    {
+        $document = $event->getSubject();
+
+        if (!is_object($document)) {
+            throw new \InvalidArgumentException(sprintf(
+                'Expected an object, got a "%s"',
+                gettype($document)
+            ));
+        }
+
+        $class = get_class($document);
+
+        $resourceMetadata = $this->registry->getByClass($class);
+        $documentMetadata = $this->documentManager->getClassMetadata(get_class($document));
+
+        $options = array_merge(
+            [
+                'default_parent_path' => null,
+                'autocreate' => false,
+            ],
+            $resourceMetadata->getParameter('options')
+        );
```

Comment:
Would be nice to be able to define the options for the driver and do some validation on them -- I guess we should then talk about creating a framework for this type of thing.